### PR TITLE
[ColorControl]Add checks for rate and step size parameters in commands handling

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
This PR adds checks to the color control cluster commands handling based on this Spec pr
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9140

- Rate, ~~RateX and RateY~~, or StepSize parameters shall not be 0. 
- Rate checks are ignored for commands executing a moveMode stop.
- Cluster revision bump in XML
- ~~Yaml test updates.~~ I will update the test-plan first
- Cluster revision bump in relevant zap files (update_cluster_revisions.py).
- regen.

**Edit: RateX and RateY = 0 remains accepted and it stops any ongoing transition**

### CODE REVIEW

File: src/app/clusters/color-control-server/color-control-server.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:1]: It's hard to review this until https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9140 gets finalized...  We should make sure that happens and that PR is merged, then figure out things here.

**Review by lpbeliveau — APPROVED** [review]

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 1]
```diff
@@ -491,7 +491,15 @@ bool ColorControlServer::stopMoveStepCommand(app::CommandHandler * commandObj, c
     EndpointId endpoint = commandPath.mEndpointId;
     Status status       = Status::Success;
 
-    if (shouldExecuteIfOff(endpoint, optionsMask, optionsOverride))
+    // StopMoveStep command has no effect on an active color loop.
+    // Fetch if it is supported and active.
+    uint8_t isColorLoopActive = 0;
+    if (ColorControlServer::Instance().HasFeature(endpoint, ColorControlServer::Feature::kColorLoop))
+    {
+        Attributes::ColorLoopActive::Get(endpoint, &isColorLoopActive);
```
Comment (andy31415) [inline src/app/clusters/color-control-server/color-control-server.cpp:499]: should we check return codes on these getters?

**Review by andy31415 — COMMENTED** [review]
@jmartinez-silabs where are the test changes?

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp:499]: We can, what should we do on failures? return a command failure? What IM error code?

Comment (andy31415) [reply src/app/clusters/color-control-server/color-control-server.cpp:499]: I do not actually know ... but at least log and set some random true/false default. Silent ignore seems bad.

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 2]
```diff
@@ -491,7 +491,15 @@ bool ColorControlServer::stopMoveStepCommand(app::CommandHandler * commandObj, c
     EndpointId endpoint = commandPath.mEndpointId;
     Status status       = Status::Success;
 
-    if (shouldExecuteIfOff(endpoint, optionsMask, optionsOverride))
+    // StopMoveStep command has no effect on an active color loop.
+    // Fetch if it is supported and active.
+    uint8_t isColorLoopActive = 0;
+    if (ColorControlServer::Instance().HasFeature(endpoint, ColorControlServer::Feature::kColorLoop))
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:497]: There should be a test for this.  Probably a certification test, ideally, so followup is OK.

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 3]
```diff
@@ -1669,8 +1678,8 @@ bool ColorControlServer::stepHueCommand(app::CommandHandler * commandObj, const
     ColorHueTransitionState * colorHueTransitionState = getColorHueTransitionState(endpoint);
     VerifyOrExit(colorHueTransitionState != nullptr, status = Status::UnsupportedEndpoint);
 
-    // Confirm validity of the step mode received
-    if (stepMode == HueStepMode::kUnknownEnumValue)
+    // Confirm validity of the step mode and step size received
+    if (stepMode == HueStepMode::kUnknownEnumValue || stepSize == 0)
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:1682]: There needs to be a cert test for this.

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 4]
```diff
@@ -2364,6 +2375,12 @@ bool ColorControlServer::stepColorCommand(app::CommandHandler * commandObj, cons
     VerifyOrExit(colorXTransitionState != nullptr, status = Status::UnsupportedEndpoint);
     VerifyOrExit(colorYTransitionState != nullptr, status = Status::UnsupportedEndpoint);
 
+    if (stepX == 0 && stepY == 0)
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:2378]: Again, needs cert test.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 4 above)
Comment (jmartinez-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp:2386]: Done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 3 above)
Comment (jmartinez-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp:1686]: done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp]: done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp:497]: I will add a steps to the test plan for this then add to the certification test in the SDK
I create a ticket to keep track of it
https://github.com/project-chip/connectedhomeip/issues/34423

**Pull request merged**